src/registry.rs

```diff
@@ -56,3 +56,38 @@ pub trait Registry {
     /// An error will be returned if the content could not be written.
     fn write(&self, content: impl Read) -> Result<ContentHash, Self::Error>;
 }
+
+#[cfg(test)]
+#[expect(clippy::unwrap_used, reason = "the tests do not need error handling")]
+mod tests {
+    use std::io::{Cursor, Read};
+
+    use super::*;
+
+    pub fn read_a_written_value(registry: &impl Registry) {
+        let content_1 = b"hello";
+        let content_2 = b"world";
+
+        let hash_1 = registry.write(Cursor::new(content_1)).unwrap();
+        let hash_2 = registry.write(Cursor::new(content_2)).unwrap();
+
+        let mut read_content_1 = registry.read(hash_1).unwrap().unwrap();
+        let mut buffer_1 = Vec::new();
+        read_content_1.read_to_end(&mut buffer_1).unwrap();
```

Comment:
You can simply use [read_to_string](https://doc.rust-lang.org/std/io/trait.Read.html#method.read_to_string)